pkg/process/metadata/parser/java/util.go

```diff
@@ -0,0 +1,126 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package javaparser
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/rickar/props"
+	"github.com/vibrantbyte/go-antpath/antpath"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type mapSource struct {
+	m map[string]string
+}
+
+// Get the value for the supplied key
+func (y *mapSource) Get(key string) (string, bool) {
+	val, ok := y.m[key]
+	return val, ok
+}
```

Comment:
you're wrapping a map, and creating a wrapper to the access of the map.

Why don't you use a map from the first place?

The entire struct does not give you any improvement or simplification

Reply:
I agree that's not providing simplifications but I'm using an external library `github.com/rickar/props` that's defining the interface PropertySource having methods Get and GetDefault. This lib is also providing some accessor for env vars, args and property file, using also resolution of placeholders (`${another.prop}`) as spring boot allows. Hence I had to cope with this interface in order to use it.
Once those PropertySource are created they are added to a source of type `props.Combined` that's doing chaining